proxy/src/fully_qualified_authority.rs

```diff
@@ -5,9 +5,26 @@ use std::str::FromStr;
 
 use http::uri::Authority;
 
+/// A fully qualified authority according to Kubernetes service naming
+/// conventions.
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct FullyQualifiedAuthority(Authority);
 
+/// A fully qualified authority, and whether or not the `Destination` service
+/// should be queried for that name.
```

Comment:
This comment just repeats exactly what the code says, so I don't think we should include it, so that we don't have to keep it in sync with code changes in the future.